File: src/Reloaded.Web/ClientApp/src/app/models/reload.ts

export class Reload {
  reloadId!: number;
  accountId!: number;
  createdOn!: Date;
  lastUpdatedOn!: Date;
  nickname!: string;
  notes!: string;
  powder!: string;
  powderCharge!: number;
  overallLength!: number;
  primer = new Primer();
  casing = new Casing();
  bullet = new Bullet();
}

export class Primer {
  brand!: string;
  type!: string;
}

export class Casing {
  brand!: string;
  cartridge!: string;
  newBrass!: boolean;
}

export class Bullet {
  weight!: number;
  brand!: string;
  construction!: string;
  tipType!: string;
  baseType!: string;
  caliber!: string;
}

export class ReloadFirearmMap {
  reloadId!: number;
  firearmId!: number;
}

File: src/Reloaded.Common/Models/Reload.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Reloaded.Common.Enums.Components;
using Reloaded.Common.Models.Components;

namespace Reloaded.Common.Models
{
	[Table("reload")]
	public class Reload : BaseEntity
	{
		[Key]
		public int ReloadId { get; set; }

        public string Nickname { get; set; }

        public string Notes { get; set; }

        [Column("powderType")]
		public GunPowder Powder { get; set; } = GunPowder.None;

		[Required]
		[Column("powderChargeGr")]
		public double PowderCharge { get; set; }

		[Required]
		[Column("overallLengthIn")]
		public double OverallLength { get; set; }

        public Primer Primer { get; set; } = new Primer();

		public Brass Casing { get; set; } = new Brass();

		public Bullet Bullet { get; set; } = new Bullet();

		public List<ReloadResult> Results { get; set; } = new List<ReloadResult>();
    }
}